### PR TITLE
feat: base design system package

### PR DESCRIPTION
**Pull request by simonheys**
Sets up a base design system package `@argent/ui` with theme and example components and stories for chakra-ui. This is a starting point for further development - the theme is not yet used in the extension.

See the package README for an example: https://github.com/argentlabs/argent-x/tree/feat/init-design-system-chakra/packages/ui

### CODE REVIEW

File: packages/extension/package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
   },
   "dependencies": {
     "@argent/x-sessions": "^4.8.5",
+    "@argent-x/ui": "*",
```
Comment (janek26) [inline packages/extension/package.json:76]:
```suggestion
    "@argent/ui": "*",
```

Lets make it one argent

File: packages/extension/src/ui/App.tsx [hunk 2]
```diff
@@ -1,3 +1,9 @@
+import {
+  FixedGlobalStyle,
+  ThemeProvider,
+  ThemedGlobalStyle,
+  muiTheme,
+} from "@argent-x/ui/src/theme"
```
Comment (janek26) [inline packages/extension/src/ui/App.tsx:6]: I dont think that we should import from src in any case

There should be a build pipeline which takes care of bundling to `@argent/ui` or creates entrypoints like `@argent/ui/theme` which is also possible inside package.json using the exports syntax

File: packages/ui/package.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  "scripts": {
+    "gen:theme-typings": "chakra-cli tokens ./src/theme/chakraTheme.ts",
+    "lint": "eslint . --cache --ext .ts,.tsx",
+    "postinstall": "yarn gen:theme-typings"
+  },
+  "dependencies": {
+    "@chakra-ui/react": "^2.3.6",
+    "@emotion/react": "^11.10.4",
+    "@emotion/styled": "^11.10.4",
```
Comment (janek26) [inline packages/ui/package.json:34]: are both of the emotion libs needed?

File: packages/ui/package.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    "@google/model-viewer": "^2.0.0",
+    "@mui/icons-material": "^5.3.1",
+    "@mui/material": "^5.1.0",
+    "@mui/styled-engine-sc": "^5.10.3",
```
Comment (janek26) [inline packages/ui/package.json:38]: can we drop mui in `@argent/ui`

File: packages/ui/package.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    "@tippyjs/react": "^4.2.6",
+    "colord": "^2.9.2",
+    "framer-motion": "^7.6.1",
+    "lodash-es": "^4.17.21",
+    "qr-code-styling": "^1.6.0-rc.1",
+    "react": "^18.0.0",
```
Comment (janek26) [inline packages/ui/package.json:44]: should be peer dep

File: packages/ui/package.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    "react-copy-to-clipboard": "^5.0.4",
+    "react-dom": "^18.0.0",
```
Comment (janek26) [inline packages/ui/package.json:46]: peer dep

File: packages/ui/package.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    "react-dropzone": "^14.0.0",
+    "react-hook-form": "^7.33.0",
+    "react-measure": "^2.5.2",
+    "react-router-dom": "^6.0.1",
+    "react-select": "^5.4.0",
+    "react-textarea-autosize": "^8.3.4",
+    "styled-components": "^5.3.5",
+    "styled-normalize": "^8.0.7",
```
Comment (janek26) [inline packages/ui/package.json:54]: styled-* needed?

File: packages/ui/package.json [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    "tracking-link": "^4.0.1",
```
Comment (janek26) [inline packages/ui/package.json:55]: we have our own impl now

File: packages/ui/package.json [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    "qr-code-styling": "^1.6.0-rc.1",
+    "react": "^18.0.0",
+    "react-copy-to-clipboard": "^5.0.4",
+    "react-dom": "^18.0.0",
+    "react-dropzone": "^14.0.0",
+    "react-hook-form": "^7.33.0",
+    "react-measure": "^2.5.2",
+    "react-router-dom": "^6.0.1",
+    "react-select": "^5.4.0",
+    "react-textarea-autosize": "^8.3.4",
```
Comment (janek26) [inline packages/ui/package.json:52]: if I import a button from this package, I dont want to use all of these dependencies I dont need. I think we should put heavy focus on treeshaking here and use the package.json exports syntax to get different entrypoints

File: packages/ui/src/theme/GlobalStyle.tsx [hunk 10]
```diff
@@ -0,0 +1,49 @@
+import { createGlobalStyle } from "styled-components"
```
Comment (janek26) [inline packages/ui/src/theme/GlobalStyle.tsx:1]: I dont think this should be part of a ui library

File: packages/ui/src/theme/ThemeProvider.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { FC, PropsWithChildren } from "react"
```
Comment (janek26) [inline packages/ui/src/theme/ThemeProvider.tsx:1]: lets get rid of this file

File: packages/ui/src/theme/muiTheme.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { createTheme } from "@mui/material/styles"
```
Comment (janek26) [inline packages/ui/src/theme/muiTheme.ts:1]: lets remove this too

**Review by janek26 — APPROVED** [review]

File: packages/ui/package.json (hunk 9 above)
Comment (simonheys) [reply packages/ui/package.json]: As discussed - will come back to this laters

**Review by janek26 — APPROVED** [review]
thanks a lot for this! 
I'll add turborepo to the mix with ww

**Pull request merged**